lib/parser.dart

```diff
@@ -1229,13 +1229,30 @@ class _Parser {
         combinatorType = TokenKind.COMBINATOR_PLUS;
         break;
       case TokenKind.GREATER:
+        // Parse > or >>>
         _eat(TokenKind.GREATER);
-        combinatorType = TokenKind.COMBINATOR_GREATER;
+        if (_maybeEat(TokenKind.GREATER)) {
+          _eat(TokenKind.GREATER);
+          combinatorType = TokenKind.COMBINATOR_DEEP;
+        } else {
+          combinatorType = TokenKind.COMBINATOR_GREATER;
+        }
         break;
       case TokenKind.TILDE:
         _eat(TokenKind.TILDE);
         combinatorType = TokenKind.COMBINATOR_TILDE;
         break;
+      case TokenKind.SLASH:
+        // Parse /deep/
+        _eat(TokenKind.SLASH);
+        var ate = _maybeEat(TokenKind.IDENTIFIER);
+        var tok = ate ? _previousToken : _peekToken;
+        if (!(ate && tok.text == 'deep')) {
```

Comment:
We should support `/deep/` for backwards compatibility as up until recently `>>>` was merely an alias. We could deprecate `/deep/` and issue a warning instructing users to switch to `>>>` if desired.

I'm skeptical we'll see any combinators of this form with reserved names. The purposed [reference combinator](https://www.w3.org/TR/selectors4/#idref-combinators) used this syntax, but has since been [removed](https://drafts.csswg.org/selectors/#changes).